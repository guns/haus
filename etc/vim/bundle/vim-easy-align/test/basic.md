" :source run.vim
" @a

Paul McCartney 1942
George Harrison 1943
Ringo Starr 1940
Pete Best 1941

| Option| Type | Default | Description |
|--|--|--|--|
| threads | Fixnum | 1 | number of threads in the thread pool |
|queues |Fixnum | 1 | number of concurrent queues |
|queue_size | Fixnum | 1000 | size of each queue |
|   interval | Numeric | 0 | dispatcher interval for batch processing |
|batch | Boolean | false | enables batch processing mode |
 |batch_size | Fixnum | nil | number of maximum items to be assigned at once |
 |logger | Logger | nil | logger instance for debug logs |

aaa,   bb,c
d,eeeeeee
fffff, gggggggggg,
h, ,           ii
j,,k

```ruby

a =
a = 1
bbbb .= 2
ccccccc = 3
ccccccccccccccc
ddd = #
eeee === eee = eee = eee=f
fff = ggg += gg &&= gg
g != hhhhhhhh == # 8
i   := 5
i     %= 5
i       *= 5
j     =~ 5
j   >= 5
aa      =>         123
aa <<= 123
aa        >>= 123
bbb               => 123
c     => 1233123
d   =>      123
dddddd &&= 123
dddddd ||= 123
dddddd /= 123
gg <=> ee

options = { :caching => nil,
            :versions => 3,
            "cache=blocks" => false }.merge(options)

apple = 1 # comment not aligned
banana = 'Gros Michel' # comment 2

# let g:easy_align_delimiters = { '#': { 'pattern': '#\+', 'ignores': ['String'] } }

apple = 1 # comment not aligned
apricot = 'DAD' + 'F#AD'
banana = 'Gros Michel' # comment 2

```

```yaml

mysql:
  # JDBC driver for MySQL database:
  driver: com.mysql.jdbc.Driver
  # JDBC URL for the connection (jdbc:mysql://HOSTNAME/DATABASE)
  url: jdbc:mysql://localhost/test
  database: test
  "user:pass":r00t:pa55

```
```c

/* a */ b = c
aa >= bb
// aaa = bbb = cccc
/* aaaa = */ bbbb   === cccc   " = dddd = " = eeee
aaaaa /* bbbbb */      == ccccc /* != eeeee = */ === fffff

```

my_object
      .method1.chain
    .second_method.call
      .third.call
     .method_4.execute

my_object.
      method1.chain.
    second_method.call.
      third.call.
     method_4.execute

   | Option| Type | Default | Description |
    |--|--|--|--|
       | threads | Fixnum | 1 | number of threads in the thread pool |
  |queues |Fixnum | 1 | number of concurrent queues |
   |queue_size | Fixnum | 1000 | size of each queue |
    |   interval | Numeric | 0 | dispatcher interval for batch processing |
       |batch | Boolean | false | enables batch processing mode |
   |batch_size | Fixnum | nil | number of maximum items to be assigned at once |
   |logger | Logger | nil | logger instance for debug logs |


```c

const char* str = "Hello";
int64_t count = 1 + 2;
static double pi = 3.14;
static std::map<std::string, float>* scores =    pointer;

```

Paul McCartney 1942
George Harrison 1943
Ringo Starr 1940
Pete Best 1941

...-.-----
..--..----
.---...---
----....--
.---.....-
..--......
...-.....-

  apple = 1
    banana = 2
      cake = 3
        daisy = 4
     eggplant = 5


```c

int a = 1;
long b = 2;
float c = 3;
string d = 4;
// this line should not get aligned
long int e = 5;
std::map f;
std::map g; /* this? */
short h /* how about this? */ = 6;
string i = "asdf";

```

```ruby

a =
a = 1
bbbb .= 2
ccccccc = 3
ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
ddd = #
eeee === eee = eee = eee=f
fff = ggg += gg &&= gg
g != hhhhhhhh == # 8
i   := 5
i     %= 5
i       *= 5
j     =~ 5
j   >= 5
aa      =>         123
aa <<= 123
aa        >>= 123
bbb               => 123
c     => 1233123
d   =>      123
dddddd &&= 123
dddddd ||= 123
dddddd /= 123
gg <=> ee

a = 1
bb = 22
ccc = 333
dddd = "asdf"

```

a a a a a
bbb bbb bbb bbb bbb
ccccc ccccc ccccc ccccc ccccc
  d ddd dddd ddd d

aaaaa = 123456778901234567890 =
bbbbbbbbbb = 12345 =

aaaaa = 123456778901234567890 =
cccccccccccccccccc = 123 =

aaaaa = 123456778901234567890 =
cccccccccccccccccc = 12345678 =

aaaaa = 12345 =
bbbbbbbbbb = 123456778901234567890       =

aaaaa =             12345
bbbbbbbbbb = 123456778901234567890

123456 789
1234567890 1

1234 56
123456 7890

 aaaaa = 123456778901234567890 =
     cccccccccccccccccc =       12345678 =

```
 aaaaa =   123456778901234567890 =
cccccccccccccccccc =12345678 =

            aaaa             = 123456778901234567890 =
     cccccccccccccccccc      = 12345678 =

            aaaaaaaaaaaaa = 123456778901234567890 =
cc = 12345678 =

           aaaaaaaaaaa= 123
                       a = 123

           aaaaaaaaaaaa= 123
                       a = 123

         aaaaaaaaaaaa = 123
       aaaaaaaaaaa = 123
      aaaaaaaaaa  = 123
     aaa     = 123
    aa      = 123
   a      = 123

            aaaa = 123456778901234567890 =
cccccccccccccc = 12345678 =

              aaaa = 123456778901234567890 =
               bbbbbb = 4
cccccccccccccccccc = 12345678 =

                    aaaa = 123456778901234567890 =
     cccccccccccccccccc = 12345678 =

                    aaaaa                     = 123456778901234567890 =
                     cc                       = 12345678 =

=aaaaa = 123456778901234567890 =
=    cccccccccccccccccc = 12345678 =

```

a,,bbb
aa,,bb
aaa,,b
aaaa,,
aaa,b,
aa,bb,
a,bbb,
