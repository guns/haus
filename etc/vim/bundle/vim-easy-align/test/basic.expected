" :source run.vim
" @a

Pa  ul  Mc  Ca  rt  ne  y{  {1  }}  94 2
Ge  or  ge  Ha  rr  is  on  {{  1}  }9  43
Ri  ng  o   St  ar  r   {{  1}  }9  40
Pe  te  Be  st  {{  1}  }9  41

  Paul McCartney{{1}}942
George Harrison {{1}}943
 Ringo Starr    {{1}}940
  Pete Best     {{1}}941

  Paul McCartney 1942
George Harrison  1943
 Ringo Starr     1940
  Pete Best      1941

Paul  McCartney 1942
George Harrison 1943
Ringo     Starr 1940
Pete       Best 1941

Paul   McCartney 1942
George Harrison 1943
Ringo  Starr 1940
Pete   Best 1941

Paul McCartney  1942
George Harrison 1943
Ringo Starr     1940
Pete Best       1941

Paul   McCartney 1942
George  Harrison 1943
Ringo      Starr 1940
Pete        Best 1941

  Paul McCartney 1942
George  Harrison 1943
 Ringo     Starr 1940
  Pete      Best 1941

Paul  McCartney 1942
George Harrison 1943
Ringo     Starr 1940
Pete       Best 1941

  Paul McCartney 1942
George Harrison 1943
 Ringo Starr 1940
  Pete Best 1941

Paul   McCartney 1942
George Harrison  1943
Ringo  Starr     1940
Pete   Best      1941

Paul McCartney  1942
George Harrison 1943
Ringo Starr     1940
Pete Best       1941

Paul   McCartney 1942
George Harrison 1943
Ringo  Starr 1940
Pete   Best 1941

|     Option | Type    | Default | Description                                    |
|         -- | --      |      -- | --                                             |
|    threads | Fixnum  |       1 | number of threads in the thread pool           |
|     queues | Fixnum  |       1 | number of concurrent queues                    |
| queue_size | Fixnum  |    1000 | size of each queue                             |
|   interval | Numeric |       0 | dispatcher interval for batch processing       |
|      batch | Boolean |   false | enables batch processing mode                  |
| batch_size | Fixnum  |     nil | number of maximum items to be assigned at once |
|     logger | Logger  |     nil | logger instance for debug logs                 |

|     Option |    Type | Default |                                    Description |
|         -- |      -- |      -- |                                             -- |
|    threads |  Fixnum |       1 |           number of threads in the thread pool |
|     queues |  Fixnum |       1 |                    number of concurrent queues |
| queue_size |  Fixnum |    1000 |                             size of each queue |
|   interval | Numeric |       0 |       dispatcher interval for batch processing |
|      batch | Boolean |   false |                  enables batch processing mode |
| batch_size |  Fixnum |     nil | number of maximum items to be assigned at once |
|     logger |  Logger |     nil |                 logger instance for debug logs |

| Option     | Type    | Default | Description                                    |
| --         | --      | --      | --                                             |
| threads    | Fixnum  | 1       | number of threads in the thread pool           |
| queues     | Fixnum  | 1       | number of concurrent queues                    |
| queue_size | Fixnum  | 1000    | size of each queue                             |
| interval   | Numeric | 0       | dispatcher interval for batch processing       |
| batch      | Boolean | false   | enables batch processing mode                  |
| batch_size | Fixnum  | nil     | number of maximum items to be assigned at once |
| logger     | Logger  | nil     | logger instance for debug logs                 |

aaa,   bb,         c
d,     eeeeeee
fffff, gggggggggg,
h,     ,           ii
j,     ,           k

```ruby

              a        =
              a        =  1
              bbbb    .=  2
              ccccccc  =  3
ccccccccccccccc
              ddd      =  #
              eeee    === eee      =  eee  =  eee = f
              fff      =  ggg      += gg  &&= gg
              g       !=  hhhhhhhh == # 8
              i       :=  5
              i       %=  5
              i       *=  5
              j       =~  5
              j       >=  5
              aa      =>  123
              aa      <<= 123
              aa      >>= 123
              bbb     =>  123
              c       =>  1233123
              d       =>  123
              dddddd  &&= 123
              dddddd  ||= 123
              dddddd  /=  123
              gg      <=> ee

a               =
a               =   1
bbbb            .=  2
ccccccc         =   3
ccccccccccccccc
ddd             =   #
eeee            === eee      =  eee =   eee = f
fff             =   ggg      += gg  &&= gg
g               !=  hhhhhhhh == # 8
i               :=  5
i               %=  5
i               *=  5
j               =~  5
j               >=  5
aa              =>  123
aa              <<= 123
aa              >>= 123
bbb             =>  123
c               =>  1233123
d               =>  123
dddddd          &&= 123
dddddd          ||= 123
dddddd          /=  123
gg              <=> ee

              a   =
              a   =        1
           bbbb  .=        2
        ccccccc   =        3
ccccccccccccccc
            ddd   =        #
           eeee ===      eee  = eee = eee=f
            fff   =      ggg += gg &&= gg
              g  != hhhhhhhh == # 8
              i  :=        5
              i  %=        5
              i  *=        5
              j  =~        5
              j  >=        5
             aa  =>      123
             aa <<=      123
             aa >>=      123
            bbb  =>      123
              c  =>  1233123
              d  =>      123
         dddddd &&=      123
         dddddd ||=      123
         dddddd  /=      123
             gg <=>       ee

a         =
a         =        1
bbbb     .=        2
ccccccc   =        3
ccccccccccccccc
ddd       =        #
eeee    ===      eee  = eee   = eee = f
fff       =      ggg += gg  &&=  gg
g        != hhhhhhhh == # 8
i        :=        5
i        %=        5
i        *=        5
j        =~        5
j        >=        5
aa       =>      123
aa      <<=      123
aa      >>=      123
bbb      =>      123
c        =>  1233123
d        =>      123
dddddd  &&=      123
dddddd  ||=      123
dddddd   /=      123
gg      <=>       ee

a         =
a         = 1
bbbb     .= 2
ccccccc   = 3
ccccccccccccccc
ddd       = #
eeee    === eee       = eee   = eee = f
fff       = ggg      += gg  &&= gg
g        != hhhhhhhh == # 8
i        := 5
i        %= 5
i        *= 5
j        =~ 5
j        >= 5
aa       => 123
aa      <<= 123
aa      >>= 123
bbb      => 123
c        => 1233123
d        => 123
dddddd  &&= 123
dddddd  ||= 123
dddddd   /= 123
gg      <=> ee

a         =
a         = 1
bbbb     .= 2
ccccccc   = 3
ccccccccccccccc
ddd       = #
eeee    === eee = eee = eee=f
fff       = ggg += gg &&= gg
g        != hhhhhhhh == # 8
i        := 5
i        %= 5
i        *= 5
j        =~ 5
j        >= 5
aa       => 123
aa      <<= 123
aa      >>= 123
bbb      => 123
c        => 1233123
d        => 123
dddddd  &&= 123
dddddd  ||= 123
dddddd   /= 123
gg      <=> ee

options = {       :caching => nil,
                 :versions => 3,
            "cache=blocks" => false }.merge(options)

options = { :caching       => nil,
            :versions      => 3,
            "cache=blocks" => false }.merge(options)

options                     = { :caching => nil,
            :versions      => 3,
            "cache=blocks" => false }.merge(options)

apple = 1              # comment not aligned
banana = 'Gros Michel' # comment 2

# let g:easy_align_delimiters = { '#': { 'pattern': '#\+', 'ignores': ['String'] } }

apple = 1              # comment not aligned
apricot = 'DAD' + 'F#AD'
banana = 'Gros Michel' # comment 2

apple = 1 # comment not aligned
apricot = 'DAD' + 'F#AD'
banana = 'Gros Michel' # comment 2

```

```yaml

        mysql:
  # JDBC driver for MySQL database:
       driver: com.mysql.jdbc.Driver
  # JDBC URL for the connection (jdbc:mysql://HOSTNAME/DATABASE)
          url: jdbc:mysql://localhost/test
     database: test
  "user:pass": r00t:pa55

mysql:
  # JDBC driver for MySQL database:
  driver:      com.mysql.jdbc.Driver
  # JDBC URL for the connection (jdbc:mysql://HOSTNAME/DATABASE)
  url:         jdbc:mysql://localhost/test
  database:    test
  "user:pass": r00t:pa55

```
```c

/* a */ b          = c
aa                >= bb
// aaa             = bbb        = cccc
/* aaaa            = */ bbbb  === cccc   " = dddd   = " = eeee
aaaaa /* bbbbb */ == ccccc /*  != eeeee    = */   === fffff

/* a */ b           = c
aa                 >= bb
// aaa = bbb = cccc
/* aaaa = */ bbbb === cccc   " = dddd = "      = eeee
aaaaa /* bbbbb */  == ccccc /* != eeeee = */ === fffff

```

my_object
      .method1    .chain
    .second_method.call
      .third      .call
     .method_4    .execute

my_object
         .method1      .  chain
         .second_method.   call
         .third        .   call
         .method_4     .execute

my_object
    .      method1.chain
    .second_method.call
    .        third.call
    .     method_4.execute

my_object
         .      method1.chain
         .second_method.call
         .        third.call
         .     method_4.execute

my_object
         .      method1.  chain
         .second_method.   call
         .        third.   call
         .     method_4.execute

my_object
    .method1      .chain
    .second_method.call
    .third        .call
    .method_4     .execute

my_object
    .method1.chain
    .second_method.call
    .third.call
    .method_4.execute

my_object        .
      method1    .chain.
    second_method.call.
      third      .call.
     method_4    .execute

  | Option     |    Type | Default |                                    Description |
  | --         |      -- | --      |                                             -- |
  | threads    |  Fixnum | 1       |           number of threads in the thread pool |
  | queues     |  Fixnum | 1       |                    number of concurrent queues |
  | queue_size |  Fixnum | 1000    |                             size of each queue |
  | interval   | Numeric | 0       |       dispatcher interval for batch processing |
  | batch      | Boolean | false   |                  enables batch processing mode |
  | batch_size |  Fixnum | nil     | number of maximum items to be assigned at once |
  | logger     |  Logger | nil     |                 logger instance for debug logs |

  |     Option |    Type | Default |                                    Description |
  |         -- |      -- |      -- |                                             -- |
  |    threads |  Fixnum |       1 |           number of threads in the thread pool |
  |     queues |  Fixnum |       1 |                    number of concurrent queues |
  | queue_size |  Fixnum |    1000 |                             size of each queue |
  |   interval | Numeric |       0 |       dispatcher interval for batch processing |
  |      batch | Boolean |   false |                  enables batch processing mode |
  | batch_size |  Fixnum |     nil | number of maximum items to be assigned at once |
  |     logger |  Logger |     nil |                 logger instance for debug logs |

  | Option     | Type    | Default | Description                                    |
  | --         | --      | --      | --                                             |
  | threads    | Fixnum  | 1       | number of threads in the thread pool           |
  | queues     | Fixnum  | 1       | number of concurrent queues                    |
  | queue_size | Fixnum  | 1000    | size of each queue                             |
  | interval   | Numeric | 0       | dispatcher interval for batch processing       |
  | batch      | Boolean | false   | enables batch processing mode                  |
  | batch_size | Fixnum  | nil     | number of maximum items to be assigned at once |
  | logger     | Logger  | nil     | logger instance for debug logs                 |


  |Option         |Type        |Default     |Description                                        |
  |--             |--          |--          |--                                                 |
  |threads        |Fixnum      |1           |number of threads in the thread pool               |
  |queues         |Fixnum      |1           |number of concurrent queues                        |
  |queue_size     |Fixnum      |1000        |size of each queue                                 |
  |interval       |Numeric     |0           |dispatcher interval for batch processing           |
  |batch          |Boolean     |false       |enables batch processing mode                      |
  |batch_size     |Fixnum      |nil         |number of maximum items to be assigned at once     |
  |logger         |Logger      |nil         |logger instance for debug logs                     |

  |    Option    ~|    Type   ~|    Default~|    Description                                   ~|
  |    --        ~|    --     ~|    --     ~|    --                                            ~|
  |    threads   ~|    Fixnum ~|    1      ~|    number of threads in the thread pool          ~|
  |    queues    ~|    Fixnum ~|    1      ~|    number of concurrent queues                   ~|
  |    queue_size~|    Fixnum ~|    1000   ~|    size of each queue                            ~|
  |    interval  ~|    Numeric~|    0      ~|    dispatcher interval for batch processing      ~|
  |    batch     ~|    Boolean~|    false  ~|    enables batch processing mode                 ~|
  |    batch_size~|    Fixnum ~|    nil    ~|    number of maximum items to be assigned at once~|
  |    logger    ~|    Logger ~|    nil    ~|    logger instance for debug logs                ~|

```c

const char*                          str    = "Hello";
int64_t                              count  = 1 + 2;
static double                        pi     = 3.14;
static std::map<std::string, float>* scores = pointer;

```

  Paul@@@ McCartney{{{ 1942
George@@@ Harrison {{{ 1943
 Ringo@@@ Starr    {{{ 1940
  Pete@@@ Best     {{{ 1941

  Paul McCartney 1942
George  Harrison 1943
 Ringo     Starr 1940
  Pete      Best 1941

...    - .     -----
..    -- ..     ----
.    --- ...     ---
    ---- ....     --
.    --- .....     -
..    -- ......
...    - .....     -

...  -   .     -----
..   --  ..    ----
.   ---  ...    ---
    ---- ....   --
.   ---  .....   -
..   --  ......
...  -   .....   -

... -    .     -----
..  --   ..    ----
.   ---  ...   ---
    ---- ....  --
.   ---  ..... -
..  --   ......
... -    ..... -

...  -   .     -----
..   --  ..    ----
.   ---  ...    ---
    ---- ....   --
.   ---  .....   -
..   --  ......
...  -   .....   -

...  -   .-----
..   --  ..----
.   ---  ...---
    ---- ....--
.   ---  .....-
..   --  ......
...  -   .....-

... -    .     -----
..  --   ..    ----
.   ---  ...   ---
    ---- ....  --
.   ---  ..... -
..  --   ......
... -    ..... -

... -    .-----
..  --   ..----
.   ---  ...---
    ---- ....--
.   ---  .....-
..  --   ......
... -    .....-

...    - .-----
..    -- ..----
.    --- ...---
    ---- ....--
.    --- .....-
..    -- ......
...    - .....-

...-.-----
..--..----
.---...---
----....--
.---.....-
..--......
...-.....-

...  -   .     -----
..   --  ..    ----
.   ---  ...    ---
    ---- ....   --
.   ---  .....   -
..   --  ......
...  -   .....   -

...  -   .-----
..   --  ..----
.   ---  ...---
    ---- ....--
.   ---  .....-
..   --  ......
...  -   .....-

... -    .     -----
..  --   ..    ----
.   ---  ...   ---
    ---- ....  --
.   ---  ..... -
..  --   ......
... -    ..... -

...    - .     -----
..    -- ..     ----
.    --- ...     ---
    ---- ....     --
.    --- .....     -
..    -- ......
...    - .....     -

...-.     -----
..--..     ----
.---...     ---
----....     --
.---.....     -
..--......
...-.....     -

...    - .-----
..    -- ..----
.    --- ...---
    ---- ....--
.    --- .....-
..    -- ......
...    - .....-

        apple    = 1       = 2
        banana   = 2    = 2
        cake     = 3    = 2
        daisy    = 4 = 2
        eggplant = 5 = 2

   apple = 1 = 2
  banana = 2 = 2
    cake = 3 = 2
   daisy = 4 = 2
eggplant = 5 = 2

apple    = 1 = 2
banana   = 2 = 2
cake     = 3 = 2
daisy    = 4 = 2
eggplant = 5 = 2

apple    = 1 = 2
banana   = 2 = 2
cake     = 3 = 2
daisy    = 4 = 2
eggplant = 5 = 2

     apple = 1 = 2
    banana = 2 = 2
      cake = 3 = 2
     daisy = 4 = 2
  eggplant = 5 = 2

  apple    = 1 = 2
  banana   = 2 = 2
  cake     = 3 = 2
  daisy    = 4 = 2
  eggplant = 5 = 2

1 apple    = 1       = 2
1 banana   = 2    = 2
1 cake     = 3    = 2
1 daisy    = 4 = 2
1 eggplant = 5 = 2

apple    = 1
banana   = 2
cake     = 3
daisy    = 4
eggplant = 5

  a     pple     = 1
        banana   = 2
        cake     = 3
        daisy    = 4
        eggplant = 5

  apple     = 1
   banana   = 2
   cake     = 3
   daisy    = 4
   eggplant = 5

  apple = 1       = 2
    banana = 2    = 2
      cake = 3    = 2
        daisy = 4 = 2
     eggplant = 5 = 2

  apple       = 1
    banana    = 2
      cake    = 3
        daisy = 4
     eggplant = 5

     apple = 1
    banana = 2
      cake = 3
     daisy = 4
  eggplant = 5

           apple = 1
          banana = 2
            cake = 3
           daisy = 4
        eggplant = 5

     apple = 1
    banana = 2
      cake = 3
     daisy = 4
  eggplant = 5

        apple = 1
       banana = 2
         cake = 3
        daisy = 4
     eggplant = 5

        apple    = 1
        banana   = 2
        cake     = 3
        daisy    = 4
        eggplant = 5

        apple    = 1
        banana   = 2
        cake     = 3
        daisy    = 4
        eggplant = 5

  apple    = 1
  banana   = 2
  cake     = 3
  daisy    = 4
  eggplant = 5

  apple       = 1
    banana    = 2
      cake    = 3
        daisy = 4
     eggplant = 5


```c

                        int a = 1;
                       long b = 2;
                      float c = 3;
                     string d = 4;
// this line should not get aligned
                   long int e = 5;
                  std::map f;
std::map g; /* this? */
short h /* how about this? */ = 6;
                     string i = "asdf";

int a                         = 1;
long b                        = 2;
float c                       = 3;
string d                      = 4;
// this line should not get aligned
long int e                    = 5;
std::map f;
std::map g; /* this? */
short h /* how about this? */ = 6;
string i                      = "asdf";

```

```ruby

a         =
a         =                       1
bbbb     .=                       2
ccccccc   =                       3
                    ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
ddd       =                       #
eeee    === eee       = eee = eee=f
fff       = ggg      += gg &&=   gg
g        != hhhhhhhh ==         # 8
i        :=                       5
i        %=                       5
i        *=                       5
j        =~                       5
j        >=                       5
aa       =>                     123
aa      <<=                     123
aa      >>=                     123
bbb      =>                     123
c        =>                 1233123
d        =>                     123
dddddd  &&=                     123
dddddd  ||=                     123
dddddd   /=                     123
gg      <=>                      ee

a                    =
a                    =        1
bbbb                .=        2
ccccccc              =        3
ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
ddd                  =        #
eeee               ===      eee  = eee   = eee = f
fff                  =      ggg += gg  &&=  gg
g                   != hhhhhhhh == # 8
i                   :=        5
i                   %=        5
i                   *=        5
j                   =~        5
j                   >=        5
aa                  =>      123
aa                 <<=      123
aa                 >>=      123
bbb                 =>      123
c                   =>  1233123
d                   =>      123
dddddd             &&=      123
dddddd             ||=      123
dddddd              /=      123
gg                 <=>       ee

              a   =
              a   = 1
           bbbb  .= 2
        ccccccc   = 3
ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
            ddd   = #
           eeee === eee       = eee   = eee = f
            fff   = ggg      +=  gg &&= gg
              g  != hhhhhhhh == # 8
              i  := 5
              i  %= 5
              i  *= 5
              j  =~ 5
              j  >= 5
             aa  => 123
             aa <<= 123
             aa >>= 123
            bbb  => 123
              c  => 1233123
              d  => 123
         dddddd &&= 123
         dddddd ||= 123
         dddddd  /= 123
             gg <=> ee

a         =
a         =        1
bbbb     .=        2
ccccccc   =        3
ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
ddd       =        #
eeee    ===      eee  = eee   = eee = f
fff       =      ggg += gg  &&=  gg
g        != hhhhhhhh == # 8
i        :=        5
i        %=        5
i        *=        5
j        =~        5
j        >=        5
aa       =>      123
aa      <<=      123
aa      >>=      123
bbb      =>      123
c        =>  1233123
d        =>      123
dddddd  &&=      123
dddddd  ||=      123
dddddd   /=      123
gg      <=>       ee

              a   =
              a   =        1
           bbbb  .=        2
        ccccccc   =        3
ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
            ddd   =        #
           eeee ===      eee  = eee   = eee = f
            fff   =      ggg +=  gg &&=  gg
              g  != hhhhhhhh == # 8
              i  :=        5
              i  %=        5
              i  *=        5
              j  =~        5
              j  >=        5
             aa  =>      123
             aa <<=      123
             aa >>=      123
            bbb  =>      123
              c  =>  1233123
              d  =>      123
         dddddd &&=      123
         dddddd ||=      123
         dddddd  /=      123
             gg <=>       ee

              a   =
              a   = 1
           bbbb  .= 2
        ccccccc   = 3
ccccccccccccccc
# eeeeeeeeeeeeeeee
e # asdf
            ddd   = #
           eeee === eee = eee = eee=f
            fff   = ggg += gg &&= gg
              g  != hhhhhhhh == # 8
              i  := 5
              i  %= 5
              i  *= 5
              j  =~ 5
              j  >= 5
             aa  => 123
             aa <<= 123
             aa >>= 123
            bbb  => 123
              c  => 1233123
              d  => 123
         dddddd &&= 123
         dddddd ||= 123
         dddddd  /= 123
             gg <=> ee

a =         1
bb =       22
ccc =     333
dddd = "asdf"

a = 1
bb = 22
ccc = 333
dddd = "asdf"

   a = 1
  bb = 22
 ccc = 333
dddd = "asdf"

a    =      1
bb   =     22
ccc  =    333
dddd = "asdf"

   a =      1
  bb =     22
 ccc =    333
dddd = "asdf"

a =         1
bb =       22
ccc =     333
dddd = "asdf"

   a = 1
  bb = 22
 ccc = 333
dddd = "asdf"

```

    a a         a a         a
  bbb bbb     bbb bbb     bbb
ccccc ccccc ccccc ccccc ccccc
    d ddd    dddd ddd       d

aaaaa = 123456778901234567890 =
bbbbbbbbbb =    12345         =

aaaaa =     123456778901234567890 =
cccccccccccccccccc = 123          =

aaaaa =       123456778901234567890 =
cccccccccccccccccc = 12345678       =

aaaaa =              12345         =
bbbbbbbbbb = 123456778901234567890 =

  aaaaa    =         12345
bbbbbbbbbb = 123456778901234567890

123456    789
1234567890 1

1234    56
123456 7890

       aaaaa        = 123456778901234567890 =
 cccccccccccccccccc =       12345678        =

|   Option   |  Type   | Default |                  Description                   |
|     --     |   --    |   --    |                       --                       |
|  threads   | Fixnum  |    1    |      number of threads in the thread pool      |
|   queues   | Fixnum  |    1    |          number of concurrent queues           |
| queue_size | Fixnum  |  1000   |               size of each queue               |
|  interval  | Numeric |    0    |    dispatcher interval for batch processing    |
|   batch    | Boolean |  false  |         enables batch processing mode          |
| batch_size | Fixnum  |   nil   | number of maximum items to be assigned at once |
|   logger   | Logger  |   nil   |         logger instance for debug logs         |

|   Option   |    Type | Default |                                    Description |
|     --     |      -- |      -- |                                             -- |
|  threads   |  Fixnum |       1 |           number of threads in the thread pool |
|   queues   |  Fixnum |       1 |                    number of concurrent queues |
| queue_size |  Fixnum |    1000 |                             size of each queue |
|  interval  | Numeric |       0 |       dispatcher interval for batch processing |
|   batch    | Boolean |   false |                  enables batch processing mode |
| batch_size |  Fixnum |     nil | number of maximum items to be assigned at once |
|   logger   |  Logger |     nil |                 logger instance for debug logs |

| Option     |  Type   | Default | Description                                    |
| --         |   --    | --      | --                                             |
| threads    | Fixnum  | 1       | number of threads in the thread pool           |
| queues     | Fixnum  | 1       | number of concurrent queues                    |
| queue_size | Fixnum  | 1000    | size of each queue                             |
| interval   | Numeric | 0       | dispatcher interval for batch processing       |
| batch      | Boolean | false   | enables batch processing mode                  |
| batch_size | Fixnum  | nil     | number of maximum items to be assigned at once |
| logger     | Logger  | nil     | logger instance for debug logs                 |

  Option   | Type    | Default | Description                                    |
    --     | --      |   --    | --                                             |
 threads   | Fixnum  |    1    | number of threads in the thread pool           |
  queues   | Fixnum  |    1    | number of concurrent queues                    |
queue_size | Fixnum  |  1000   | size of each queue                             |
 interval  | Numeric |    0    | dispatcher interval for batch processing       |
  batch    | Boolean |  false  | enables batch processing mode                  |
batch_size | Fixnum  |   nil   | number of maximum items to be assigned at once |
  logger   | Logger  |   nil   | logger instance for debug logs                 |

```
 aaaaa =     123456778901234567890 =
cccccccccccccccccc =12345678       =

            aaaa             = 123456778901234567890 =
     cccccccccccccccccc      =       12345678        =

       aaaa        = 123456778901234567890 =
cccccccccccccccccc = 12345678 =

                   aaaa        = 123456778901234567890 =
            cccccccccccccccccc = 12345678 =

            aaaa        = 123456778901234567890 =
     cccccccccccccccccc = 12345678 =

            aaaa        = 123456778901234567890 =
     cccccccccccccccccc = 12345678 =

            aaaaaaaaaaaaa = 123456778901234567890 =
                 cc       = 12345678 =

                  aaaaaaaaaaa = 123
                       a      = 123

                 aaaaaaaaaaaa = 123
                       a      = 123

aaaaaaaaaaaa = 123
aaaaaaaaaaa  = 123
 aaaaaaaaaa  = 123
    aaa      = 123
     aa      = 123
     a       = 123

              aaaaaaaaaaaa = 123
              aaaaaaaaaaa  = 123
               aaaaaaaaaa  = 123
                  aaa      = 123
                   aa      = 123
                   a       = 123

         aaaaaaaaaaaa = 123
         aaaaaaaaaaa  = 123
          aaaaaaaaaa  = 123
             aaa      = 123
              aa      = 123
              a       = 123

            aaaa      = 123456778901234567890 =
       cccccccccccccc = 12345678 =

                aaaa        = 123456778901234567890 =
               bbbbbb       = 4
         cccccccccccccccccc = 12345678 =

                    aaaa        = 123456778901234567890 =
             cccccccccccccccccc = 12345678 =

                    aaaaa = 123456778901234567890 =
                     cc   = 12345678 =

=          aaaaa        = 123456778901234567890 =
=    cccccccccccccccccc = 12345678       =

=aaaaa =           123456778901234567890 =
=    cccccccccccccccccc = 12345678       =

```

 a,    ,   bbb
 aa,   ,   bb
aaa,   ,    b
aaaa,  ,
aaa,   b,
 aa,  bb,
 a,   bbb,

1    22222  33  444 555   6666  7    888
11    222  3333   4 55    6666 77    888
111   22   333  444 55555 6666  7  88888
1111   2    33  444 555    66  777     8

1 22222   33 444 555   6666  7  888
11 222  3333  4  55    6666 77  888
111 22   333 444 55555 6666  7  88888
1111 2    33 444 555     66 777 8

1 22222 33  444  555  6666  7   888
11 222 3333   4  55   6666 77   888
111 22 333  444 55555 6666  7  88888
1111 2 33   444  555   66  777   8

